src/table/dtable.jl

```diff
@@ -19,12 +19,32 @@ the underlying partitions was applied to it (currently only `filter`).
 mutable struct DTable
     chunks::VTYPE
     tabletype
-    DTable(chunks::VTYPE, tabletype) = new(chunks, tabletype)
+    schema::Union{Nothing, Tables.Schema}
+    DTable(chunks::VTYPE, tabletype) = new(chunks, tabletype, nothing)
+    DTable(chunks::VTYPE, tabletype, schema) = new(chunks, tabletype, schema)
 end
 
 DTable(chunks::Vector{Dagger.EagerThunk}, args...) = DTable(VTYPE(chunks), args...)
 DTable(chunks::Vector{Dagger.Chunk}, args...) = DTable(VTYPE(chunks), args...)
 
+"""
+    DTable(table; tabletype=nothing) -> DTable
+
+Constructs a `DTable` using a `Tables.jl` compatible `table` input.
+Calls `Tables.partitions` on the `table` and assumes the provided partitioning.
+"""
+function DTable(table; tabletype=nothing)
+    chunks = Vector{Dagger.Chunk}()
+    type = nothing
+    sink = Tables.materializer(tabletype !== nothing ? tabletype() : partition)
+    for partition in Tables.partitions(table)
+        tpart = sink(partition)
+        push!(chunks, Dagger.tochunk(tpart))
+        type === nothing && (type = typeof(tpart).name.wrapper;)
```

Comment:
```suggestion
        if type === nothing
            type = typeof(tpart).name.wrapper
        end
```

Makes it clear that you're not trying to construct a `NamedTuple` :smile: